torch/nn/modules/activation.py

```diff
@@ -1162,7 +1162,11 @@ def forward(
             )
             # We have to use list comprehensions below because TorchScript does not support
             # generator expressions.
-            if torch.overrides.has_torch_function(tensor_args):
+            torch_fn_mode_stack = torch.overrides._get_current_function_mode_stack()
+            is_pre_dispatch_tracing = len(torch_fn_mode_stack) == 1 and \
+                type(torch_fn_mode_stack[0]) == torch.fx.experimental.proxy_tensor.PreDispatchTorchFunctionMode
```

Comment:
This part (where you test that there is only one function mode and it is this), should get factored into a helper function. This is too many lines here.